dimitrov/Sem04/src/main/kotlin/ru/bmstu/iu4/Sem04/createTable/functionForBuild.kt

```diff
@@ -6,69 +6,42 @@ fun String.newMaxHeight(width: Int): Int {
     return this.chunked(width).lastIndex
 }
 
-/**
- * Сложение строк с пробелами
- */
-/*fun listChunked(text: String, width: Int): List<String> {
-    val temp = text.split(" ").toMutableList()
-    println("input string: $temp")
-    val tempList = mutableListOf<String>()
-    tempList.add(temp.first())
-    println("first in tempList: $tempList")
-    temp.removeAt(index = 0)
-    println("remove 0 in temp: $temp")
-    for ((index, element) in temp.withIndex()) {
-
-
-        /*if (tempList[index].length + temp[index].length + 1 <= width) {
-            tempList[index] = tempList[index].plus(" ").plus(element)
-            temp.removeAt(index)
-        }
-        else tempList.add(element)*/
-
-
-        println("delete in circle: $temp")
-        println("add in circle: $tempList")
-    }
-    return tempList
-} */
-
-fun listChunked(text: String, width: Int): List<String> {
-    return text.chunked(width)
-}
-
 fun maxWidth(vararg strings: String, width: Int): Int {
     var tempMaxWidth = width
-    for (element in strings) {
-        val temp = element.split(" ")
-        for (items in temp) if (items.length > tempMaxWidth) tempMaxWidth = items.length
+    strings.forEach {

```

Comment:
strings.maxBy { string -> 
    string.split(" ").maxBy { it.length }?.length ?: -1 
}